### ISSUE TITLE
[redbean] on Windows, get 403 forbidden

### ISSUE THREAD

**Issue by jgranduel** [jart/cosmopolitan#1075]
Hi, 

thanks again and deeply for this outstanding project ! All my best wishes for Jart, all maintainers and best success to the project itself!

versions: cosmos-3.1.3, cosmos-3.2 on Windows-10

I tried to add the simplest html page to redbean as described on [redbean landing page](https://redbean.dev/#usage):
```
> echo '<b>hello</b>' >hello.html
> zip redbean.com hello.html

> ./redbean.com
```
But the hello.html sends a 403 status. Do you know why (and maybe add a comment on redbean documentation...)?

**Comment by pkulchenko** [comment]
I can't reproduce getting 403 on the current master. There are only two places in redbean returning 403, so it would be useful to see the following:
1. The content of .init.lua (if it's not empty)
2. The exact URL of the request you're sending
3. The log file content, as it should show warnings for those 403 responses returned.

**Comment by jgranduel** [comment]
Thanks for your answer.

Here's how I've been doing. 
I just downloaded cosmos-3.2.zip and expanded it. So all settings are default ones. Folder is in my HOME, not in `C:/bin` if has some effects.
```
cosmos-3.2\bin> .\redbean.com
I2024-01-08T19:01:30.557510:tool/net/redbean.c:7105:redbean:5796] (srvr) listen http://xxx:8080
I2024-01-08T19:01:30+000403:tool/net/redbean.c:7105:redbean:5796] (srvr) listen http://127.0.0.1:8080
```
In Firefox:
clicking on hello.html (or `curl http://127.0.0.1:8080/hello.html`)


I2024-01-08T19:01:34.861009:tool/net/redbean.c:6023:redbean:23892] (req) received 127.0.0.1:51661 HTTP11 GET http://127.0.0.1:8080/ "" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0"
I2024-01-08T19:01:37.466176:tool/net/redbean.c:6023:redbean:23892] (req) received 127.0.0.1:51661 HTTP11 GET http://127.0.0.1:8080/hello.html "http://127.0.0.1:8080/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0"
W2024-01-08T19:01:37+000712:tool/net/redbean.c:6080:redbean:23892] (srvr) asset "/hello.html" 0100600 isn't readable
E2024-01-08T19:01:37+000676:tool/net/redbean.c:2576:redbean:23892] (srvr) server error: 403 Forbidden "/hello.html"
>:
I2024-01-08T19:02:18.784384:tool/net/redbean.c:7130:redbean:5796] (srvr) received SIGINT
W2024-01-08T19:02:18+002945:tool/net/redbean.c:1353:redbean:5796] (stat) 23892 terminated with SIGINT (0 workers remain)
I2024-01-08T19:02:18+000591:tool/net/redbean.c:7137:redbean:5796] (srvr) shutdown complete
```

1. I unzipped .init.lua like so:
```
 .\unzip.com -p .\redbean.com .init.lua
-- special script called by main redbean process at startup
HidePath('/usr/share/zoneinfo/')
HidePath('/usr/share/ssl/')
```

2. url is http://127.0.0.1:8080/hello.html

3. The log file content, as it should show warnings for those 403 responses returned
I did it like so:
```

**Comment by pkulchenko** [comment]
> (srvr) asset "/hello.html" 0100600 isn't readable

I wonder what permissions the file has in the file system and, as the result, in the zip archive. redbean requires "other" permissions on the files it serves, even if they are stored in the archive for consistency with externally served files, and none are set in this case.

**Comment by jgranduel** [comment]
Thanks.
I don't know how to set full permissions in Windows. I tried from WSL. I chmod 777 hello.html, rezipped it into redbean. But no success yet. 🤔.  If third column in redbean landing page are permissions, all the files I have been adding so far have 0100600 permissions.

**Comment by pkulchenko** [comment]
>  If third column in redbean landing page are permissions, all the files I have been adding so far have 0100600 permissions.

correct, those are all permissions. Interesting, I'd expect it to work. It shows for me as 100664 on Windows.

**Comment by jgranduel** [comment]
Can you tell me exactly what you did? I tried from a Windows Terminal with PowerShell-7.4. Why would our permissions be different??

**Comment by pkulchenko** [comment]
Not sure; I used `winrar a -afzip redbean.com hello.html` command and it stored the file with the expected permissions.

**Comment by jgranduel** [comment]
I never used _winrar_.  What would be the equivalent with _zip_ utility which comes along with cosmos?
I tried again with a new file
````
Pwsh-7.4>"hi" > foo.txt
> .\zip.com  redbean.com baz.txt
  adding: baz.txt (stored 0%)
```
and `foo.txt`  has 100600 permission status.

Instead of PowerShell, I tried in bash that comes with cosmos-3.2. Same result.

**Comment by jgranduel** [comment]
Hi, 

I dit another test today... And all worked out as expected : 
```
> unzip.exe cosmos-3.2.zip bin/redbean
> mv .\redbean redbean.com
> "hi" > a.txt 
> zip redbean.com a.txt
```
and all went well. This time permission was set to '100644'. A PowerShell issue 🤨?
Is there a way to change permission once in redbean?

Thanks.

**Comment by moshen** [comment]
I just ran into this trying to serve a folder locally on Windows 10 in Powershell

```
scripts\redbean.com -D .\gh-pages\                                                                                                                                                                   
I2025-03-03T09:48:44.425467:tool/net/redbean.c:1122:redbean:15224] (cfg) program directory: ./gh-pages/
I2025-03-03T09:48:44+001045:tool/net/redbean.c:7052:redbean:15224] (srvr) listen http://172.29.176.1:8080
I2025-03-03T09:48:44+000557:tool/net/redbean.c:7052:redbean:15224] (srvr) listen http://172.21.128.1:8080
I2025-03-03T09:48:44+000433:tool/net/redbean.c:7052:redbean:15224] (srvr) listen http://172.27.64.1:8080
I2025-03-03T09:48:44+000360:tool/net/redbean.c:7052:redbean:15224] (srvr) listen http://192.168.1.112:8080
I2025-03-03T09:48:44+000333:tool/net/redbean.c:7052:redbean:15224] (srvr) listen http://127.0.0.1:8080
I2025-03-03T09:48:44+000358:tool/net/redbean.c:7052:redbean:15224] (srvr) listen http://172.29.240.1:8080
I2025-03-03T09:48:48.914716:tool/net/redbean.c:6112:redbean:38736] (req) received 127.0.0.1:61331 HTTP11 GET http://127.0.0.1:8080/ "" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/133.0.0.0 Safari/537.36"
W2025-03-03T09:48:48+000424:tool/net/redbean.c:6170:redbean:38736] (srvr) asset "/index.html" 0100600 isn't readable
E2025-03-03T09:48:48+000195:tool/net/redbean.c:2635:redbean:38736] (srvr) server error: 403 Forbidden "/index.html"
>:
I2025-03-03T10:00:18.011695:tool/net/redbean.c:7077:redbean:15224] (srvr) received SIGINT
I2025-03-03T10:00:18+000195:tool/net/redbean.c:7085:redbean:15224] (srvr) shutdown complete
```

**Comment by moshen** [comment]
When I switched to the redbean version from https://cosmo.zip/pub/cosmos/bin/ the issue appears to be resolved for me

**Comment by Skhmt** [comment]
> When I switched to the redbean version from https://cosmo.zip/pub/cosmos/bin/ the issue appears to be resolved for me

This fixed the issue for me too
